airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py

```diff
@@ -383,6 +392,21 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         state_dict[self.cursor_field] = datetime.strftime(max(current_parsed_datetime, latest_record_datetime), self.datetime_format_string)
 
         state_dict["schema"] = self._get_schema_map()
+
+        history = current_stream_state.get("history", {})
+
+        # add record to history if record modified date in range delta start from state
+        if latest_record_datetime.date() + timedelta(days=self.days) >= self._get_datetime_from_stream_state(state_dict).date():
+            history_item = set(history.setdefault(latest_record_datetime.strftime("%Y-%m-%d"), set()))
+            history_item.add(latest_record["_ab_source_file_url"])
+            history[latest_record_datetime.strftime("%Y-%m-%d")] = history_item
+
+        # reset history to new date state
+        if current_parsed_datetime.date() != self._get_datetime_from_stream_state(state_dict).date():
+            history = {date: history[date] for date in history if datetime.strptime(date, "%Y-%m-%d").date() + timedelta(days=self.days) >= self._get_datetime_from_stream_state(state_dict).date()}
```

Comment:
Assuming we always run on the same server, I guess timezone problems can be ignored

Reply:
This code solve how to clean history from old file which `creation_date` is less than `state + delta`.